Compilation fails with LibXL 3.8.4.0

Compilation seems to fail with the newest version of LibXL. Compiles fine with LibXL 3.8.3.0.

Tested both with php-7.2.13 and php-7.1.25

```
[root@devel php-7.2.13]# cd ext/excel/
[root@devel excel]# module load php72
[root@devel excel]# git checkout php7
Branch php7 set up to track remote branch php7 from origin.
Switched to a new branch 'php7'
[root@devel excel]# phpize
Configuring for:
PHP Api Version:         20170718
Zend Module Api No:      20170718
Zend Extension Api No:   320170718
[root@devel excel]# ./configure --with-excel --with-libxl-incdir=/usr/lib64/libxl-3.8.4.0/include_c --with-libxl-libdir=/usr/lib64/libxl-3.8.4.0/lib64 --with-libxml-dir=/usr/include/libxml2
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for a sed that does not truncate output... /bin/sed
checking for cc... cc
checking for C compiler default output file name... a.out
checking whether the C compiler works... yes
checking whether we are cross compiling... no
checking for suffix of executables... 
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether cc accepts -g... yes
checking for cc option to accept ISO C89... none needed
checking how to run the C preprocessor... cc -E
checking for icc... no
checking for suncc... no
checking whether cc understands -c and -o together... yes
checking for system library directory... lib
checking if compiler supports -R... no
checking if compiler supports -Wl,-rpath,... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking for PHP prefix... /opt/remi/php72/root/usr
checking for PHP includes... -I/opt/remi/php72/root/usr/include/php -I/opt/remi/php72/root/usr/include/php/main -I/opt/remi/php72/root/usr/include/php/TSRM -I/opt/remi/php72/root/usr/include/php/Zend -I/opt/remi/php72/root/usr/include/php/ext -I/opt/remi/php72/root/usr/include/php/ext/date/lib
checking for PHP extension directory... /opt/remi/php72/root/usr/lib64/php/modules
checking for PHP installed headers prefix... /opt/remi/php72/root/usr/include/php
checking if debug is enabled... no
checking if zts is enabled... no
checking for re2c... no
configure: WARNING: You will need re2c 0.13.4 or later if you want to regenerate PHP parsers.
checking for gawk... gawk
checking whether to enable excel support... yes, shared
checking C include dir for libxl... yes, shared
checking lib dir for libxl... yes, shared
checking libxml2 install dir... yes, shared
checking for excel includes... found in /usr/lib64/libxl-3.8.4.0/include_c
checking for excel libraries... found in /usr/lib64/libxl-3.8.4.0/lib64
checking for libxml2... found in /usr/include/libxml2
checking for xlCreateBookCA in -lxl... yes
checking for xlBookSetKeyA in -lxl... yes
checking for a sed that does not truncate output... (cached) /bin/sed
checking for fgrep... /bin/grep -F
checking for ld used by cc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1966080
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for ar... ar
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from cc object... ok
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if cc supports -fno-rtti -fno-exceptions... no
checking for cc option to produce PIC... -fPIC -DPIC
checking if cc PIC flag -fPIC -DPIC works... yes
checking if cc static flag -static works... no
checking if cc supports -c -o file.o... yes
checking if cc supports -c -o file.o... (cached) yes
checking whether the cc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
configure: creating ./config.status
config.status: creating config.h
config.status: executing libtool commands
[root@devel excel]# make
/bin/sh /root/php-7.2.13/ext/excel/libtool --mode=compile cc  -I. -I/root/php-7.2.13/ext/excel -DPHP_ATOM_INC -I/root/php-7.2.13/ext/excel/include -I/root/php-7.2.13/ext/excel/main -I/root/php-7.2.13/ext/excel -I/opt/remi/php72/root/usr/include/php -I/opt/remi/php72/root/usr/include/php/main -I/opt/remi/php72/root/usr/include/php/TSRM -I/opt/remi/php72/root/usr/include/php/Zend -I/opt/remi/php72/root/usr/include/php/ext -I/opt/remi/php72/root/usr/include/php/ext/date/lib -I/usr/lib64/libxl-3.8.4.0/include_c -I/usr/include/libxml2  -DHAVE_CONFIG_H  -g -O2   -c /root/php-7.2.13/ext/excel/excel.c -o excel.lo 
libtool: compile:  cc -I. -I/root/php-7.2.13/ext/excel -DPHP_ATOM_INC -I/root/php-7.2.13/ext/excel/include -I/root/php-7.2.13/ext/excel/main -I/root/php-7.2.13/ext/excel -I/opt/remi/php72/root/usr/include/php -I/opt/remi/php72/root/usr/include/php/main -I/opt/remi/php72/root/usr/include/php/TSRM -I/opt/remi/php72/root/usr/include/php/Zend -I/opt/remi/php72/root/usr/include/php/ext -I/opt/remi/php72/root/usr/include/php/ext/date/lib -I/usr/lib64/libxl-3.8.4.0/include_c -I/usr/include/libxml2 -DHAVE_CONFIG_H -g -O2 -c /root/php-7.2.13/ext/excel/excel.c  -fPIC -DPIC -o .libs/excel.o
/root/php-7.2.13/ext/excel/excel.c: In function ‘zim_ExcelSheet_insertRow’:
/root/php-7.2.13/ext/excel/excel.c:2622: error: too few arguments to function ‘xlSheetInsertRowA’
/root/php-7.2.13/ext/excel/excel.c: In function ‘zim_ExcelSheet_insertCol’:
/root/php-7.2.13/ext/excel/excel.c:2630: error: too few arguments to function ‘xlSheetInsertColA’
/root/php-7.2.13/ext/excel/excel.c: In function ‘zim_ExcelSheet_removeRow’:
/root/php-7.2.13/ext/excel/excel.c:2638: error: too few arguments to function ‘xlSheetRemoveRowA’
/root/php-7.2.13/ext/excel/excel.c: In function ‘zim_ExcelSheet_removeCol’:
/root/php-7.2.13/ext/excel/excel.c:2646: error: too few arguments to function ‘xlSheetRemoveColA’
/root/php-7.2.13/ext/excel/excel.c: In function ‘zim_ExcelSheet_table’:
/root/php-7.2.13/ext/excel/excel.c:4669: warning: passing argument 3 of ‘xlSheetTableA’ from incompatible pointer type
/usr/lib64/libxl-3.8.4.0/include_c/SheetA.h:179: note: expected ‘int *’ but argument is of type ‘zend_long *’
/root/php-7.2.13/ext/excel/excel.c:4669: warning: passing argument 4 of ‘xlSheetTableA’ from incompatible pointer type
/usr/lib64/libxl-3.8.4.0/include_c/SheetA.h:179: note: expected ‘int *’ but argument is of type ‘zend_long *’
/root/php-7.2.13/ext/excel/excel.c:4669: warning: passing argument 5 of ‘xlSheetTableA’ from incompatible pointer type
/usr/lib64/libxl-3.8.4.0/include_c/SheetA.h:179: note: expected ‘int *’ but argument is of type ‘zend_long *’
/root/php-7.2.13/ext/excel/excel.c:4669: warning: passing argument 6 of ‘xlSheetTableA’ from incompatible pointer type
/usr/lib64/libxl-3.8.4.0/include_c/SheetA.h:179: note: expected ‘int *’ but argument is of type ‘zend_long *’
/root/php-7.2.13/ext/excel/excel.c:4669: warning: passing argument 7 of ‘xlSheetTableA’ from incompatible pointer type
/usr/lib64/libxl-3.8.4.0/include_c/SheetA.h:179: note: expected ‘int *’ but argument is of type ‘zend_long *’
/root/php-7.2.13/ext/excel/excel.c:4669: warning: passing argument 8 of ‘xlSheetTableA’ from incompatible pointer type
/usr/lib64/libxl-3.8.4.0/include_c/SheetA.h:179: note: expected ‘int *’ but argument is of type ‘zend_long *’
/root/php-7.2.13/ext/excel/excel.c: In function ‘zim_ExcelFilterColumn_getCustomFilter’:
/root/php-7.2.13/ext/excel/excel.c:5193: warning: passing argument 3 of ‘xlFilterColumnGetCustomFilterA’ from incompatible pointer type
/usr/lib64/libxl-3.8.4.0/include_c/FilterColumnA.h:23: note: expected ‘const char **’ but argument is of type ‘char **’
/root/php-7.2.13/ext/excel/excel.c:5193: warning: passing argument 5 of ‘xlFilterColumnGetCustomFilterA’ from incompatible pointer type
/usr/lib64/libxl-3.8.4.0/include_c/FilterColumnA.h:23: note: expected ‘const char **’ but argument is of type ‘char **’
make: *** [excel.lo] Error 1
```

--- a/excel.c
+++ b/excel.c
@@ -2607,6 +2607,19 @@ EXCEL_METHOD(Sheet, writeCol)
 		RETURN_BOOL(xlSheet ## func_name (sheet, r, c));	\
 	}
 
+#define PHP_EXCEL_SHEET_GET_BOOL_STATE_384(func_name) \
+	{	\
+		SheetHandle sheet;	\
+		zval *object = getThis();	\
+		long r, c;	\
+		zend_bool u = 1; \
+		if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "ll", &r, &c, &u) == FAILURE) {	\
+			RETURN_FALSE;	\
+		}	\
+		SHEET_FROM_OBJECT(sheet, object);	\
+		RETURN_BOOL(xlSheet ## func_name (sheet, r, c, u));	\
+	}
+
 /* {{{ proto bool ExcelSheet::isFormula(int row, int column)
 	Determine if the cell contains a formula */
 EXCEL_METHOD(Sheet, isFormula)
@@ -2645,35 +2658,51 @@ EXCEL_METHOD(Sheet, isDate)
 }
 /* }}} */
 
-/* {{{ proto bool ExcelSheet::insertRow(int row_first, int row_last)
+/* {{{ proto bool ExcelSheet::insertRow(int row_first, int row_last, bool update_named_ranges)
 	Inserts rows from rowFirst to rowLast */
 EXCEL_METHOD(Sheet, insertRow)
 {
+#if LIBXL_VERSION >= 0x03080301
+	PHP_EXCEL_SHEET_GET_BOOL_STATE_384(InsertRow)
+#else
 	PHP_EXCEL_SHEET_GET_BOOL_STATE(InsertRow)
+#endif
 }
 /* }}} */
 
-/* {{{ proto bool ExcelSheet::insertCol(int col_first, int col_last)
+/* {{{ proto bool ExcelSheet::insertCol(int col_first, int col_last, bool update_named_ranges)
 	Inserts columns from colFirst to colLast */
 EXCEL_METHOD(Sheet, insertCol)
 {
+#if LIBXL_VERSION >= 0x03080301
+	PHP_EXCEL_SHEET_GET_BOOL_STATE_384(InsertCol)
+#else
 	PHP_EXCEL_SHEET_GET_BOOL_STATE(InsertCol)
+#endif
 }
 /* }}} */
 
-/* {{{ proto bool ExcelSheet::removeRow(int row_first, int row_last)
+/* {{{ proto bool ExcelSheet::removeRow(int row_first, int row_last, bool update_named_ranges)
 	Removes rows from rowFirst to rowLast */
 EXCEL_METHOD(Sheet, removeRow)
 {
+#if LIBXL_VERSION >= 0x03080301
+	PHP_EXCEL_SHEET_GET_BOOL_STATE_384(RemoveRow)
+#else
 	PHP_EXCEL_SHEET_GET_BOOL_STATE(RemoveRow)
+#endif
 }
 /* }}} */
 
-/* {{{ proto bool ExcelSheet::removeCol(int col_first, int col_last)
+/* {{{ proto bool ExcelSheet::removeCol(int col_first, int col_last, bool update_named_ranges)
 	Removes columns from colFirst to colLast */
 EXCEL_METHOD(Sheet, removeCol)
 {
+#if LIBXL_VERSION >= 0x03080301
+	PHP_EXCEL_SHEET_GET_BOOL_STATE_384(RemoveCol)
+#else
 	PHP_EXCEL_SHEET_GET_BOOL_STATE(RemoveCol)
+#endif
 }
 /* }}} */
 
@@ -5255,24 +5284,36 @@ PHP_EXCEL_ARGINFO
 ZEND_BEGIN_ARG_INFO_EX(arginfo_Sheet_insertRow, 0, 0, 2)
 	ZEND_ARG_INFO(0, row_first)
 	ZEND_ARG_INFO(0, row_last)
+#if LIBXL_VERSION >= 0x03080301
+	ZEND_ARG_INFO(0, update_named_ranges)
+#endif
 ZEND_END_ARG_INFO()
 
 PHP_EXCEL_ARGINFO
 ZEND_BEGIN_ARG_INFO_EX(arginfo_Sheet_insertCol, 0, 0, 2)
 	ZEND_ARG_INFO(0, col_first)
 	ZEND_ARG_INFO(0, col_last)
+#if LIBXL_VERSION >= 0x03080301
+	ZEND_ARG_INFO(0, update_named_ranges)
+#endif
 ZEND_END_ARG_INFO()
 
 PHP_EXCEL_ARGINFO
 ZEND_BEGIN_ARG_INFO_EX(arginfo_Sheet_removeRow, 0, 0, 2)
 	ZEND_ARG_INFO(0, row_first)
 	ZEND_ARG_INFO(0, row_last)
+#if LIBXL_VERSION >= 0x03080301
+	ZEND_ARG_INFO(0, update_named_ranges)
+#endif
 ZEND_END_ARG_INFO()
 
 PHP_EXCEL_ARGINFO
 ZEND_BEGIN_ARG_INFO_EX(arginfo_Sheet_removeCol, 0, 0, 2)
 	ZEND_ARG_INFO(0, col_first)
 	ZEND_ARG_INFO(0, col_last)
+#if LIBXL_VERSION >= 0x03080301
+	ZEND_ARG_INFO(0, update_named_ranges)
+#endif
 ZEND_END_ARG_INFO()
 
 PHP_EXCEL_ARGINFO

